### ISSUE TITLE
Chapter 7 - Extending the database

### ISSUE THREAD

**Issue by lionel-m** [contao/docs-archive#60]
The paragraph "Extending the database" should be updated.

**Comment by aschempp** [comment]
Will you create an example or a pull request?

**Comment by lionel-m** [comment]
Yes

**Comment by lionel-m** [comment]
How to alter the database with Contao 3 for creating a new table ? We can always create a file database.sql but is this the right way ?

We assume that we have no extension...

**Comment by lionel-m** [comment]
Extending the database

The database configuration is located in the DCA files of each Contao modules. 

Each field includes an 'sql' reference. They are not sent to the database but used to calculate the difference between the Contao specifications and the actual database tables. Therefore, you can also alter fields by extending a DCA (see next chapter) or you can create a new table by creating a database.sql file under `config/database.sql`. Add the following code to create the new table:

CREATE TABLE `tl_new_table`(
  `aField` varchar(8) NOT NULL default ''
) ENGINE=MyISAM DEFAULT CHARSET=utf8;

Use the Contao install tool to update your database.

**Comment by aschempp** [comment]
Uhm, in Contao 3 we do no longer "require" the `database.sql`, the definition should be in the DCA file.

**Comment by lionel-m** [comment]
For a new table we need to create a new file for example in the `core/dca` folder ?

Example : 

``` PHP
$GLOBALS['TL_DCA']['tl_my_table'] = array
(

    // Config
    'config' => array
    (
        'sql' => array
        (
            'keys' => array
            (
                'id' => 'primary'
            )
        )
    ),

    // Fields
    'fields' => array
    (
        'id' => array
        (
            'sql'                     => "int(10) unsigned NOT NULL auto_increment"
        ),
        'pid' => array
        (
            'sql'                     => "int(10) unsigned NOT NULL default '0'"
        ),
        'tstamp' => array
        (
            'sql'                     => "int(10) unsigned NOT NULL default '0'"
        )
    )
);
```

![capture decran 2013-08-21 a 17 51 58](https://f.cloud.github.com/assets/814476/1002261/083416e4-0a7a-11e3-8520-a76b6d639262.png)

**Comment by lionel-m** [comment]
In this case :

---

Extending the database

The database configuration is located in the DCA files of each Contao modules.

Each field includes an 'sql' reference. They are not sent to the database but used to calculate the difference between the Contao specifications and the actual database tables. Therefore, you can also alter fields by extending a DCA file (see the next chapter) or you can create a new table by creating a new DCA file.

Use the Contao install tool to update your database.

---

**Comment by aschempp** [comment]
Has this been fixed in #79 or is there still something to improve?

**Comment by lionel-m** [comment]
No, it's ok.
